### ISSUE TITLE
Don't replace existing attribute

### ISSUE THREAD

**Issue by ephetic** [davesnx/babel-plugin-transform-react-qa-classes#17]
Given

```javascript
class componentWithExistingAttribute extends Component {
  render() {
    return <div data-qa="hello">
        <h1>Hello world</h1>
      </div>;
  }
}

class componentWithDyanmicAttribute extends Component {
  render() {
    const qa = "hello"
    return <div data-qa={qa}>
        <h1>Hello world</h1>
      </div>;
  }
}
```

I'd expect neither of these to have the transform applied (or at least to have the attribute applied earlier so that it gets shadowed by the explicit attribute).

```
       class componentWithExistingAttribute extends Component {
         render() {
      -    return <div data-qa="hello" data-qa="component-with-existing-attribute">
      +    return <div data-qa="hello">
               <h1>Hello world</h1>
             </div>;
         }
       }

       class componentWithDynamicAttribute extends Component {
         render() {
           const qa = "hello";
      -    return <div data-qa={qa} data-qa="component-with-dynamic-attribute">
      +    return <div data-qa={qa}>
               <h1>Hello world</h1>
             </div>;
         }
       }
```

**Comment by ephetic** [comment]
Simply changing the `.push` to `.unshift` accomplishes the shadowing solution.

**Comment by davesnx** [comment]
That  makes total sense...

**Comment by chiplay** [comment]
We'd also like to see this change, to be able to have explicit "overrides" for when components are collapsing to the same div in a way that promotes the wrong component name. Would you accept a PR with this simple change?

**Comment by davesnx** [comment]
Hey @chiplay, thanks for pushing this forward.

Don't see the use-case of this yet. Having two attributes with the same name seems not how should work.

Maybe having `data-qa="data-qa component-with-dynamic-attribute"`?

**Comment by shepitchakpavlo** [comment]
#29

**Comment by davesnx** [comment]
Published in 1.5.0
